src/vs/workbench/browser/parts/editor/editorCommandsContext.ts

```diff
@@ -0,0 +1,200 @@
+
+	// If the command arguments contain an editor context, use it
+	for (const arg of filteredArgs) {
+		if (isEditorCommandsContext(arg)) {
+			return arg;
+		}
+	}
+
+	const editorService = accessor.get(IEditorService);
+
+	// Otherwise, try to find the editor group by the URI of the resource
+	for (const uri of filteredArgs as URI[]) {
+		const editorIdentifiers = editorService.findEditors(uri);
+		if (editorIdentifiers.length) {
+			return editorIdentifiers[0];
+		}
+	}
+
+	const listService = accessor.get(IListService);
+	const editorGroupsService = accessor.get(IEditorGroupsService);
+
+	// If there is no context in the arguments, try to find the context from the focused list
+	// if the action was executed from a list
+	if (isListAcion) {
+		const list = listService.lastFocusedList as List<any>;
+		for (const focusedElement of list.getFocusedElements()) {
+			if (isGroupOrEditor(focusedElement)) {
+				return groupOrEditorToEditorContext(focusedElement, undefined, editorGroupsService);
+			}
+		}
+	}
+
+	return undefined;
+}
+
+function getMultiSelectContext(accessor: ServicesAccessor, editorContext: IEditorCommandsContext, isListAction: boolean): IEditorCommandsContext[] {
+	const listService = accessor.get(IListService);
+	const editorGroupsService = accessor.get(IEditorGroupsService);
+
+	// If the action was executed from a list, return all selected editors
+	if (isListAction) {
+		const list = listService.lastFocusedList as List<any>;
+		const selection = list.getSelectedElements().filter(isGroupOrEditor);
+
+		if (selection.length > 1) {
+			return selection.map(e => groupOrEditorToEditorContext(e, editorContext.preserveFocus, editorGroupsService));
+		}
+	}
+	// Check editors selected in the group (tabs)
+	else {
+		const group = editorGroupsService.getGroup(editorContext.groupId);
+		const editor = editorContext.editorIndex !== undefined ? group?.getEditorByIndex(editorContext.editorIndex) : group?.activeEditor;
+		// If the editor is selected, return all selected editors otherwise only use the editors context
+		if (group && editor && group.isSelected(editor)) {
+			return group.selectedEditors.map(editor => groupOrEditorToEditorContext({ editor, groupId: group.id }, editorContext.preserveFocus, editorGroupsService));
+		}
+	}
+
+	// Otherwise go with passed in context
+	return [editorContext];
+}
+
+function groupOrEditorToEditorContext(element: IEditorIdentifier | IEditorGroup, preserveFocus: boolean | undefined, editorGroupsService: IEditorGroupsService): IEditorCommandsContext {
+	if (isEditorGroup(element)) {
+		return { groupId: element.id, editorIndex: undefined, preserveFocus };
+	}
+
+	const group = editorGroupsService.getGroup(element.groupId);
+
+	return { groupId: element.groupId, editorIndex: group ? group.getIndexOfEditor(element.editor) : -1, preserveFocus };
+}
+
+function isGroupOrEditor(element: any): element is IEditorIdentifier | IEditorGroup {
```

Comment:
I would use `unknown` instead of `any` so that you know you are checking for all cases.